src/server/models/tag.ts

```diff
@@ -0,0 +1,65 @@
+import Sequelize from 'sequelize'
+import { sequelize } from '../util/sequelize'
+import { IdType } from '../../types/server/models'
+import { TAG_KEY_REGEX, TAG_STRING_REGEX } from '../../shared/util/validation'
+
+export interface TagType extends IdType, Sequelize.Model {
+  readonly tagString: string
+  readonly tagKey: string
+}
+
+// For sequelize define
+type TagTypeStatic = typeof Sequelize.Model & {
+  new (values?: object, options?: Sequelize.BuildOptions): TagType
+}
+
+export const Tag = <TagTypeStatic>sequelize.define(
+  'tag',
+  {
+    tagString: {
+      type: Sequelize.STRING(255),
+      allowNull: false,
+      unique: true,
+      validate: {
+        is: TAG_STRING_REGEX,
+      },
+    },
+    tagKey: {
+      type: Sequelize.STRING(255),
+      allowNull: false,
+      validate: {
+        is: TAG_KEY_REGEX,
```

Comment:
should we validate the lengths for `tagString` and `tagKey` as well? i'm also wondering if it helps (if it's even possible) to have a custom function to validate that `tagKey` is indeed the lowercase version of `tagString`

Reply:
I think we don't have to do these now

Reply:
ok will just KIV this then!